src/components/Waveform/Waveform.tsx

```diff
@@ -409,7 +418,7 @@ export const Waveform = forwardRef<IWaveformRef, IWaveform>((props, ref) => {
 
     const tracePlaybackValue = onCurrentDuration(data => {
       if (data.playerKey === `PlayerFor${path}`) {
-        setCurrentProgress(data.currentDuration);
+        setCurrentProgress(Number(data.currentDuration));
```

Comment:
@nilesh-simform  Can we check here if conversion is right, if not set progress to zero.
```suggestion
const convertedDuration = Number(data.currentDuration);

// Check if the conversion is valid (not NaN)
if (!isNaN(convertedDuration)) {
  setCurrentProgress(convertedDuration); // Set the progress if the conversion is valid
} else {
  setCurrentProgress(0); // Set the progress to 0 if the conversion is not valid
}
```